### ISSUE TITLE
Duration Is Incorrect in VS Test Explorer

### ISSUE THREAD

**Issue by aolszowka** [nunit/nunit3-vs-adapter#581]
| Product       | Version                  |
|---------------|--------------------------|
| NUnit         | 3.11.0                   |
| NUnit Adapter | 3.12.0                   |
| Visual Studio | 2017 Enterprise (15.8.7) |
| .NET          | 4.6.1                    |

Minimal Sample:

```csharp
namespace NUnit3ParallelTestRunnerBug
{
    using NUnit.Framework;

    [Parallelizable(ParallelScope.Children)]
    public class ParallelRunnerBug
    {
        [Test]
        public void TestCase1()
        {
            // Sleep 10 Seconds
            System.Threading.Thread.Sleep(10000);
            Assert.True(true);
        }

        [Test]
        public void TestCase2()
        {
            // Sleep 2 Seconds
            System.Threading.Thread.Sleep(2000);
            Assert.True(true);
        }

        [Test]
        public void TestCase3()
        {
            // Sleep 3 Seconds
            System.Threading.Thread.Sleep(3000);
            Assert.True(true);
        }

        [Test]
        public void TestCase4()
        {
            // Sleep 5 Seconds
            System.Threading.Thread.Sleep(5000);
            Assert.True(true);
        }
    }
}
```

The Test Explorer will show this run as taking 20 seconds; however if you look at the output you will see that it indeed ran in parallel and only took 10 seconds.

Test Explorer
![image](https://user-images.githubusercontent.com/2592881/50353682-3c3bb780-0506-11e9-80e5-0f773e6fc554.png)

Output From Runner
```
[12/21/2018 9:41:36 AM Informational] ------ Run test started ------
[12/21/2018 9:41:36 AM Informational] NUnit Adapter 3.12.0.0: Test execution started
[12/21/2018 9:41:36 AM Informational] Running selected tests in C:\Users\aceo\source\repos\NUnit3ParallelTestRunnerBug\bin\Debug\NUnit3ParallelTestRunnerBug.dll
[12/21/2018 9:41:36 AM Informational]    NUnit3TestExecutor converted 4 of 4 NUnit test cases
[12/21/2018 9:41:46 AM Informational] NUnit Adapter 3.12.0.0: Test execution complete
[12/21/2018 9:41:46 AM Informational] ========== Run test finished: 4 run (0:00:10.6590413) ==========
```

Looking at this it seems like Visual Studio is just naively adding up the total run times which is probably all they can do? 

This is probably `notabug` but it would be nice if we could create a wiki page for `Known Issues` and include this example on there. If you do not have a problem with me doing this I will take the lead on creating that documentation.

**Comment by kendrahavens** [comment]
This is a Visual Studio Test Explorer behavior and not unique to NUnit.

This is currently by design. Reporting the “real” test run time can fluctuate greatly depending on parallel test runs and ongoing processes. We also don’t include “overhead” times such as sending tests to the test frameworks to execute, receiving the response, and displaying the new tests. The time that shows is the simplistically added durations for each individual test as reported by the framework. The user development we've done on this indicated the simplistically added display is more likely to help users identify perf issues on a class/project basis.

This has come up in the [past](https://developercommunity.visualstudio.com/content/problem/190150/duration-displayed-within-test-explorer-are-incorr.html). Feel free to comment or open a new issue. I'd appreciate hearing more on your scenario if you find this is blocking you!

**Comment by aolszowka** [comment]
I mean based on this comment on the linked Developer Community page:

> The timing displayed is the time reported by the underlying test framework on how long it took the framework to execute the test. We just take that time and show it in TE window.

They're seeming to claim that the Test Adapter is at fault; at the end of the day it does not block us but is more an FYI we were concerned because we (originally) thought that the tests were NOT running in parallel via the VS Test Adapter prior to closer inspection.

Again I will close this as notabug but at least we have an audit trail on it now.

**Comment by kendrahavens** [comment]
Ah, we didn't mean to communicate the Test Adapters at fault in anyway! Only that the way it appears today is more likely to help users doing perf investigations. :)
Thank you for the discussion. If you have any future suggestions on a more comprehensive UI and what scenarios it would facilitate I'd love to hear them.

**Comment by chm-tm** [comment]
I believe that [this report](https://developercommunity.visualstudio.com/content/problem/545910/mstest-data-driven-test-duration-incorrectly-summa.html) is related. I dared to comment on another kind of parallel NUnit test in that report.
